src/classes/ADDR_Addresses_UTIL.cls

```diff
@@ -0,0 +1,618 @@
+        addr.Geolocation__Longitude__s = (decimal)sobjSrc.get(strFieldPrefixSrc + 'Longitude');
+        
+        // try to find state and country by code if they aren't specified?
+        if (isStateCountryPicklistsEnabled) {
+
+            // lookup country if needed, by country code
+            string strCountryCode = (string)sobjSrc.get(strFieldPrefixSrc + 'CountryCode');
+            if (addr.MailingCountry__c == null && strCountryCode != null) {
+                addr.MailingCountry__c = validCountriesByCode.get(strCountryCode);
+            }
+            
+            // lookup state if needed, by country code and state code.
+            string strStateCode = (string)sobjSrc.get(strFieldPrefixSrc + 'StateCode');
+            if (addr.MailingState__c == null && strCountryCode != null && strStateCode != null) {
+                // assume it's a valid statecode
+                addr.MailingState__c = strStateCode;
+            }
+        }
+        
+        // if specified, also include Address Type
+        if (strAddrType != null) 
+            addr.Address_Type__c = strAddrType;
+    }
+
+    /*******************************************************************************************************
+    * @description Utility to copy Address fields from an Address object to a Contact or Account.
+    * Handles instances where State and Country picklists are enabled, and multiline street addresses.
+    * @param addr the Address object to copy from
+    * @param sobjDst the destination Contact or Account
+    * @param strFieldPrefix the address fields to copy to, ie., Mailing, Other, Shipping, Billing
+    * @param strFieldAddrType an optional Address Type field on sobjDst to copy to 
+    * @return void
+    */ 
+    public static void copyAddressAddrSObj(Address__c addr, SObject sobjDst, string strFieldPrefix, string strFieldAddrType) {
+        sobjDst.put(strFieldPrefix + 'Street', getMultilineStreet(addr));
+        sobjDst.put(strFieldPrefix + 'City', addr.MailingCity__c);
+        sobjDst.put(strFieldPrefix + 'PostalCode', addr.MailingPostalCode__c);        
+        sobjDst.put(strFieldPrefix + 'Latitude', addr.Geolocation__Latitude__s);
+        sobjDst.put(strFieldPrefix + 'Longitude', addr.Geolocation__Longitude__s); 
+        sobjDst.put('Current_Address__c', addr.Id);
+        
+        if (!isStateCountryPicklistsEnabled) {
+            sobjDst.put(strFieldPrefix + 'State', addr.MailingState__c);
+            sobjDst.put(strFieldPrefix + 'Country', addr.MailingCountry__c);
+        } else {
+            if (addr.MailingCountry__c != null) {
+                if (validCountriesByLabel.containsKey(addr.MailingCountry__c)) {
+                    sobjDst.put(strFieldPrefix + 'Country', addr.MailingCountry__c);
+                    sobjDst.put(strFieldPrefix + 'CountryCode', validCountriesByLabel.get(addr.MailingCountry__c));
+                } else if (validCountriesByCode.containsKey(addr.MailingCountry__c.toUpperCase())) {
```

Comment:
Nesting too many `IF` statements affects readability and makes maintenance more difficult. Consider refactoring to simplify the flow of this code block.